jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/events/JobSelectedEvent.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.workbench.es.client.editors.events;
+
+import org.jboss.errai.common.client.api.annotations.Portable;
+
+@Portable
+public class JobSelectedEvent {
+
+    private String deploymentId;
+    private Long jobId;
+    private String serverTemplateId;
+
+    public JobSelectedEvent() {
+    }
+
+    public JobSelectedEvent(String serverTemplateId,
+                            String deploymentId,
+                            Long jobId) {
+        this.serverTemplateId = serverTemplateId;
+        this.jobId = jobId;
+        this.deploymentId = deploymentId;
+    }
+
+    public String getDeploymentId() {
+        return deploymentId;
+    }
+
+    public Long getJobId() {
+        return jobId;
+    }
+
+    public String getServerTemplateId() {
+        return serverTemplateId;
+    }
+
+    @Override
+    public String toString() {
+        return "JobSelectedEvent{ " +
+                "serverTemplateId=" + serverTemplateId + " jobId=" + jobId + " deploymentId=" + deploymentId + '}';
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidMultipleUnaryOperators")
+    public int hashCode() {
```

Comment:
Given how important this `~~` thing is in GWT, I keep wondering how come we haven't already created utility classes for this. Something similar to [`Objects.hash(Object... values)`](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash(java.lang.Object...)) from the JDK.

This entire method would then just be `return Objects.hash(this.serverTemplateId, this.deploymentId, this.jobId);`

@manstis @jhrcek Do you think that this would be useful? If so, where would you place such a utility method, so that we can get everyone using it?

Anyway, just a suggestion, shouldn't block this PR.

Reply:
Sounds reasonable to me. @manstis Any suggestion where such utility should be placed? Probably somewhere in uberfire..

Reply:
@triceo oks, thanks for reviewing!

Reply:
+1